src/m365/outlook/commands/message/message-remove.ts

```diff
@@ -0,0 +1,149 @@
+  public get description(): string {
+    return 'Removes a specific message from a mailbox';
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+    this.#initTypes();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        userId: typeof args.options.userId !== 'undefined',
+        userName: typeof args.options.userName !== 'undefined',
+        force: !!args.options.force
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-i, --id <id>'
+      },
+      {
+        option: '--userId [userId]'
+      },
+      {
+        option: '--userName [userName]'
+      },
+      {
+        option: '-f, --force'
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.userId && !validation.isValidGuid(args.options.userId)) {
+          return `The value '${args.options.userId}' for 'userId' option is not a valid GUID`;
+        }
+
+        if (args.options.userName && !validation.isValidUserPrincipalName(args.options.userName)) {
+          return `The value '${args.options.userName}' for 'userName' option is not a valid user principal name`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initTypes(): void {
+    this.types.string.push('id', 'userId', 'userName');
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    const isAppOnlyAccessToken: boolean | undefined = accessToken.isAppOnlyAccessToken(auth.service.accessTokens[this.resource].accessToken);
+    let principalUrl = '';
+
+    if (isAppOnlyAccessToken) {
+      if (!args.options.userId && !args.options.userName) {
+        throw `The option 'userId' or 'userName' is required when removing a message using application permissions`;
+      }
+
+      if (args.options.userId && args.options.userName) {
+        throw `Both options 'userId' and 'userName' cannot be set when removing a message using application permissions`;
+      }
+
+      principalUrl += `users/${args.options.userId || formatting.encodeQueryParameter(args.options.userName!)}`;
+    }
+    else {
+      if (args.options.userId && args.options.userName) {
+        throw `Both options 'userId' and 'userName' cannot be set when removing a message using delegated permissions`;
+      }
+
+      if (args.options.userId || args.options.userName) {
+        principalUrl += `users/${args.options.userId || formatting.encodeQueryParameter(args.options.userName!) }`;
+      }
+      else {
+        principalUrl += 'me';
+      }
+    }
+
+    const removeMessage = async (): Promise<void> => {
+      try {
+        if (this.verbose) {
+          await logger.logToStderr(`Removing message with id ${args.options.id} using ${isAppOnlyAccessToken ? 'application' : 'delegated'} permissions`);
```

Comment:
Let's also add a period here at the end of the sentence and enclose the id with quote signs `'`.